### PR TITLE
support private classes

### PR DESCRIPTION
**Pull request by fommil**
#19 // @pjrt

### CODE REVIEW

File: src/main/scala/stalactite/deriving.scala [hunk 1]
```diff
@@ -166,7 +166,15 @@ class DerivingMacros(val c: Context) {
 
     annottees.map(_.tree) match {
       case (data: ClassDef) :: Nil =>
-        val companion = atPos(data.pos)(q"object ${data.name.toTermName} {}")
+        val mods =
+          if (data.mods.hasFlag(Flag.PRIVATE))
+            Modifiers(Flag.PRIVATE, data.mods.privateWithin)
+          else if (data.mods.hasFlag(Flag.PROTECTED))
+            Modifiers(Flag.PRIVATE, data.mods.privateWithin)
```
Comment (fommil) [inline src/main/scala/stalactite/deriving.scala]: well, that's a typo...

File: src/test/scala/stalactite/classes.scala [hunk 2]
```diff
@@ -29,6 +29,9 @@ object Bar {
   def hello: String = ""
 }
 
+@deriving(Cofoo)
+private[examples] final case class Par(s: String)
```
Comment (pjrt) [inline src/test/scala/stalactite/classes.scala]: What about the case of:

```scala
private class X
object X {
}
```
?

Comment (fommil) [reply src/test/scala/stalactite/classes.scala]: the code I added only kicks in when there is no `object`. If somebody has a private class and public companion, then they'll get a compile failure when they try to put any implicits on it.

Comment (pjrt) [reply src/test/scala/stalactite/classes.scala]: That doesn't sound right (or I'm not understanding this). 

> the code I added only kicks in when there is no object

Huh? So we can't have a companion object anymore? That doesn't sound right.

> If somebody has a private class and public companion, then they'll get a compile failure when they try to put any implicits on it.

I guess that's fine. I don't see anyone ever doing that. I mean they *may* create private instances in their public companion but that's such an odd case.

Comment (fommil) [reply src/test/scala/stalactite/classes.scala]: if you have a manually added companion, then we don't generate one.

```scala
      case (data: ClassDef) :: Nil => // just a class or trait
      case (data: ClassDef) :: (companion: ModuleDef) :: Nil => // class/trait AND companion
      case (obj: ModuleDef) :: Nil => // just an object (no class or trait)
```

**Pull request merged**